tests/algo/TestAlgo.cpp

```diff
@@ -90,10 +87,40 @@ TEST(SlackNodeAlgo, equivalent) {
   std::shared_ptr<dfl::inputs::Node> slack_node;
   dfl::algo::SlackNodeAlgorithm algo(slack_node);
 
-  for (auto it = nodes.begin(); it != nodes.end(); ++it) {
-    algo(*it);
-  }
+  std::for_each(nodes.begin(), nodes.end(), algo);
 
   ASSERT_NE(nullptr, slack_node);
   ASSERT_EQ("4", slack_node->id);  // first found
 }
+
+TEST(Connexity, base) {
+  std::vector<std::shared_ptr<dfl::inputs::Node>> nodes{
+      std::make_shared<dfl::inputs::Node>("0", 0.0), std::make_shared<dfl::inputs::Node>("1", 1.0), std::make_shared<dfl::inputs::Node>("2", 2.0),
+      std::make_shared<dfl::inputs::Node>("3", 3.0), std::make_shared<dfl::inputs::Node>("4", 5.0), std::make_shared<dfl::inputs::Node>("5", 5.0),
+      std::make_shared<dfl::inputs::Node>("6", 0.0),
+  };
+  std::vector<dfl::inputs::Node::NodeId> expected_nodes{"0", "1", "2", "3"};
+
+  nodes[0]->neighbours.push_back(nodes[1]);
+  nodes[0]->neighbours.push_back(nodes[2]);
+  nodes[2]->neighbours.push_back(nodes[3]);
+  nodes[1]->neighbours.push_back(nodes[0]);
+  nodes[2]->neighbours.push_back(nodes[0]);
+  nodes[3]->neighbours.push_back(nodes[2]);
+
+  nodes[4]->neighbours.push_back(nodes[5]);
+  nodes[5]->neighbours.push_back(nodes[6]);
+  nodes[5]->neighbours.push_back(nodes[4]);
+  nodes[6]->neighbours.push_back(nodes[5]);
+
+  dfl::algo::ConnexityAlgorithm::ConnexGroup main;
+  dfl::algo::ConnexityAlgorithm algo(main);
+
+  std::for_each(nodes.begin(), nodes.end(), algo);
+
+  ASSERT_EQ(4, main.size());
+  std::vector<dfl::inputs::Node::NodeId> nodeids_main;
+  std::for_each(main.begin(), main.end(), [&nodeids_main](const std::shared_ptr<dfl::inputs::Node>& node) { nodeids_main.push_back(node->id); });
```

Comment:
I would like to see:
- unit test with same number of nodes on right or left (@marcochiaramello is there a property that we could use as a tie -breaker in this case?
- a scaling unit test with a large complete graph

Reply:
- what do you mean same number of nodes on right or left ?
- As I don't understand quite how the IIDM is built, I need a translation of a "large complete graph" in terms of nodes (id + nominal voltage)

Reply:
- I mean 2 or more subgraphs with the same number of node
- I was thinking about a theoretical graph that has a large number of nodes and connections (a complete graph is a graph for which all nodes are connected to all others nodes, but I do agree it is not really representative in this case). We could define a nrt with the IIDM of one of the testcases of Marco and Quentin so that is is more meaningful. Let's drop this one for now.

Reply:
Also could you please add later the possibility to display the wall time of the algorithms?